main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -28,59 +31,179 @@ import scala.collection.mutable
+    val filesTest = getAllFilesWithExt(getTestDirectory(path), "flix")
+    sourcePaths = filesHere ++ filesSrc ++ filesTest
+
+    timestamps = timestamps ++ (flixPackagePaths ++ mavenPackagePaths ++ sourcePaths).map(f => f -> f.toFile.lastModified).toMap
+
+    Ok(sourcePaths ++ flixPackagePaths ++ mavenPackagePaths)
   }
 
-  private def folderMode(): Unit = { // TODO: Probably return Result or Validation
+  private def folderMode(path: Path): Result[List[Path], BootstrapError] = {
     // 1. Add *.flix, src/**.flix and test/**.flix
+    val filesHere = getAllFlixFilesHere(path)
+    val filesSrc = getAllFilesWithExt(getSourceDirectory(path), "flix")
+    val filesTest = getAllFilesWithExt(getTestDirectory(path), "flix")
+    sourcePaths = filesHere ++ filesSrc ++ filesTest
+
     // 2. Grab all jars in lib/
+    val jarFilesLib = getAllFilesWithExt(getLibraryDirectory(path), "jar")
+    mavenPackagePaths = jarFilesLib
+
     // 3. Grab all flix packages in lib/
+    val flixFilesLib = getAllFilesWithExt(getLibraryDirectory(path), "fpkg")
+    flixPackagePaths = flixFilesLib
+
+    Ok(sourcePaths ++ flixPackagePaths ++ mavenPackagePaths)
   }
 
-  def reconfigureFlix(flix: Flix): Unit = { // TODO: Probably return Result or Validation
+  def reconfigureFlix(flix: Flix): Unit = {
+    val previousSources = timestamps.keySet
+
+    for (path <- sourcePaths if hasChanged(path)) {
+      flix.addSourcePath(path)
+    }
 
-    val manifest: Manifest = ??? // TODO: Probably from a field.
+    for (path <- flixPackagePaths if hasChanged(path)) {
+      flix.addSourcePath(path)
+    }
 
-    // TODO: Add logic to check if the file has changed.
-    for (path <- sourcePaths) {
+    for (path <- mavenPackagePaths if hasChanged(path)) {
       flix.addSourcePath(path)
     }
 
-    for (flixPackagePath <- manifest.getFlixPackages) {
-      flix.addSourcePath(flixPackagePath)
+    val currentSources = (sourcePaths ++ flixPackagePaths ++ mavenPackagePaths).filter(p => Files.exists(p))
+
+    val deletedSources = previousSources -- currentSources
+    for (path <- deletedSources) {
+      flix.remSourceCode(path.toString)
     }
 
-    for (mavenJarPath <- manifest.getMavenPackages) {
-      flix.addSourcePath(mavenJarPath)
+    timestamps = currentSources.map(f => f -> f.toFile.lastModified).toMap
+  }
+
+  /**
+    * Returns true if the timestamp of the given source file has changed since the last reload
+    */
+  private def hasChanged(file: Path) = {
+    !(timestamps contains file) || (timestamps(file) != file.toFile.lastModified())
+  }
+
+  /**
+    * Returns all .flix files directly in the directory given by `p`.
+    */
+  private def getAllFlixFilesHere(path: Path): List[Path] = {
+    val files = path.toFile.listFiles()
+    if (files == null) {
+      List.empty
+    } else {
+      files.toList.map(f => f.toPath).filter(p => p.getFileName.toString.endsWith(s".flix"))
     }
+  }
 
+  //Below: copied from Packager.scala
```

Comment:
I would move these functions into the `Bootstrap` object.
Delete them from Packager.scala
And make their uses in Packager.scala point here.

The goal is that eventually we will merge packager into this class.